### PR TITLE
PWN-5181 - Add phone not found error state

### PR DESCRIPTION
**Pull request by davran1001**
…phone number not exists

## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5181

## Description of Work

Refactor general error scree, and error new error state

## Screenshots (Optional)

Screenshot #1 | Screenshot #2 | Screenshot #3
:------------:|:-------------:|:------------:
[Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3]

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/RestoreWalletInteractor.kt [hunk 1]
```diff
@@ -40,4 +40,6 @@ class RestoreWalletInteractor(
     fun setIsRestoreWalletRequestSent(isSent: Boolean) {
         customShareRestoreInteractor.setIsRestoreWalletRequestSent(isSent)
     }
+
+    fun isDeviceShareSaved(): Boolean = userRestoreInteractor.isDeviceShareSaved()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/RestoreWalletInteractor.kt:44]: use signUpDetailsStorage here too

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt [hunk 2]
```diff
@@ -59,28 +60,74 @@ class OnboardingGeneralErrorFragment :
...
+                with(buttonSecondaryFirst) {
+                    text = getString(R.string.onboarding_general_error_starting_screen_button_title)
+                    setOnClickListener { popAndReplaceFragment(OnboardingRootFragment.create(), inclusive = true) }
+                    isVisible = true
                 }
-                buttonRestoreByGoogle.setOnClickListener {
-                    presenter.useGoogleAccount()
+            }
+            is GeneralErrorScreenError.AccountNotFound -> {
+                textViewErrorTitle.text = errorState.title
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt:97]: extract to a method

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt [hunk 3]
```diff
@@ -59,28 +60,74 @@ class OnboardingGeneralErrorFragment :
...
+                textViewErrorSubtitle.text = errorState.message
+                with(buttonRestoreByGoogle) {
+                    text = if (errorState.isDeviceShareExists) getString(R.string.restore_continue_with_google) else
+                        getString(R.string.restore_another_phone_number)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt:101]: braces for if

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt [hunk 4]
```diff
@@ -59,28 +60,74 @@ class OnboardingGeneralErrorFragment :
...
+                    icon = if (errorState.isDeviceShareExists) ResourcesCompat.getDrawable(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt:102]: getDrawableCompat

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt [hunk 5]
```diff
@@ -11,14 +11,17 @@ import kotlinx.coroutines.launch
 import org.p2p.wallet.auth.interactor.restore.RestoreWalletInteractor
 import org.p2p.wallet.auth.model.OnboardingFlow
 import org.p2p.wallet.auth.model.PhoneNumber
+import org.p2p.wallet.auth.ui.generalerror.GeneralErrorScreenError
+import org.p2p.wallet.common.ResourcesProvider
 
-private const val MAX_PHONE_NUMBER_TRIES = 4
+private const val MAX_PHONE_NUMBER_TRIES = 80
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt:17]: damn, is it really 80?

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt [hunk 6]
```diff
@@ -116,6 +119,35 @@ class PhoneNumberEnterPresenter(
                 Timber.e(gatewayServiceError)
                 view?.showUiKitSnackBar(messageResId = R.string.error_too_often_otp_requests_message)
             }
+            is GatewayServiceError.PhoneNumberNotExists -> {
+                Timber.e(gatewayServiceError)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt:123]: extract to a method

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/GeneralErrorScreenError.kt [hunk 7]
```diff
@@ -17,4 +17,10 @@ sealed class GeneralErrorScreenError(
         titleResId = R.string.error_wallet_not_found_title,
         messageResId = R.string.error_wallet_not_found_message
     )
+
+    data class AccountNotFound(
+        val isDeviceShareExists: Boolean,
+        val title: String,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/GeneralErrorScreenError.kt:23]: title and message is set only once and have only one value
you can set title and message inside the GeneralErrorFragment

File: app/src/main/res/layout/fragment_onboarding_general_error.xml [hunk 8]
```diff
@@ -79,81 +79,56 @@
...
-                android:id="@+id/button_restore_with_phone"
-                style="@style/UiKit.Components.Button.Large.Outlined"
-                android:layout_width="match_parent"
-                android:layout_height="@dimen/ui_kit_button_large_height"
-                android:layout_marginBottom="8dp"
-                android:text="@string/restore_phone_number"
-                android:textColor="@color/snow"
-                app:strokeColor="@color/snow" />
+        <org.p2p.uikit.components.UiKitButton
+            android:id="@+id/button_primary_second"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_onboarding_general_error.xml:110]: camelCase for all the view ids

File: app/src/main/res/layout/fragment_onboarding_general_error.xml [hunk 9]
```diff
@@ -79,81 +79,56 @@
...
+            app:iconTint="@null" />
+
+        <org.p2p.uikit.components.UiKitButton
+            android:id="@+id/button_primary_first"
+            style="@style/UiKit.Components.Button.Large.Outlined"
+            android:layout_width="match_parent"
+            android:layout_height="@dimen/ui_kit_button_large_height"
+            android:layout_marginBottom="8dp"
+            android:text="@string/restore_phone_number"
+            android:textColor="@color/snow"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_onboarding_general_error.xml:105]: text_snow

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/res/values/onboarding_strings.xml [hunk 10]
```diff
@@ -97,9 +97,13 @@
     <string name="restore_choose_way">Choose the way\nto continue</string>
     <string name="restore_continue_with_google">Continue with Google</string>
     <string name="restore_phone_number">Continue using a phone number</string>
+    <string name="restore_another_phone_number">User an another phone number</string>
     <string name="restore_seed_phrase">Use a seed phrase</string>
+    <string name="restore_no_wallet_found_with_no_device_share_message">$s not works.\nUse an another phone number</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/onboarding_strings.xml:101]: %1$s?

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt (hunk 5 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/auth/ui/phone/PhoneNumberEnterPresenter.kt:17]: Oh, it was testing counter x D

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/GeneralErrorScreenError.kt (hunk 7 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/auth/ui/generalerror/GeneralErrorScreenError.kt:23]: it does not have one value ))

File: app/src/main/res/layout/fragment_onboarding_general_error.xml (hunk 8 above)
Comment (davran1001) [reply app/src/main/res/layout/fragment_onboarding_general_error.xml:110]: i thought it was snake case x D

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/GeneralErrorScreenError.kt (hunk 7 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/auth/ui/generalerror/GeneralErrorScreenError.kt:23]: you fill in the value depending on the isDeviceShareExists flag
so check it on GeneralErrorScreen

**Pull request merged**